sdk/core/test/integration/otp/index.test.ts

```diff
@@ -1,417 +1,426 @@
 import 'mocha'
-
 import '../env'
 
 import { expect } from 'chai'
+import cryptoRandomString from 'crypto-random-string'
+
 import { CommonNetworkMember } from '../../../src/CommonNetworkMember'
 import { SdkOptions } from '../../../src/dto/shared.dto'
+import SdkError from '../../../src/shared/SdkError'
 
-import { getOtp, generateUsername, generateEmail, getOptionsForEnvironment } from '../../helpers'
+import { generateUsername, getOptionsForEnvironment } from '../../helpers'
 import { MessageParameters } from '../../../dist/dto'
-import { TestEmailHelper } from '../../helpers/getOtpFromTestMailApp'
+import { TestmailHelper } from '../../helpers/TestmailHelper'
 
-const { TEST_SECRETS } = process.env
-const { COGNITO_PASSWORD } = JSON.parse(TEST_SECRETS)
+const { COGNITO_PASSWORD } = JSON.parse(process.env.TEST_SECRETS)
 
 const options: SdkOptions = getOptionsForEnvironment()
+const { env } = getOptionsForEnvironment(true)
+
+const prepareOtpMessageParameters = (
+  testId: string,
+  { suffix }: { suffix?: string } = {},
+): [string, string, MessageParameters] => {
+  const tag = generateTag(testId, suffix)
+  const email = TestmailHelper.generateEmailForTag(tag)
```

Comment:
I actually thought about it, but since it doesn't change the complexity in the test themselves, I don't think it's better (or worse) to do it that way. I'd rather have these actions separated and have small helpers on top of the file. 
I don't like parsing things unnecessary, because it can make email generation rigid and also raise "wtf?" questions when we come back to the code later.

Reply:
> but since it doesn't change the complexity in the test themselves

But it does! Right now, tests have to concern themselves with implementation details of TestMailHelper; they have to know that in addition to email there is some "tag" used to access messages.

On the other hand, we could avoid parsing things by storing the tag in the `Mailbox` object, as proposed in other comment...

Reply:
Oh, I meant the specs, not the helpers around it. In combination with your mailbox proposal, it makes sense to move all these helpers into one place. I will try to do that, thank you

Reply:
Isn't this the test spec? :)
https://github.com/affinityproject/affinidi-core-sdk/blob/012bcd9b27a0054030f4197512a63910d4521456/sdk/core/test/integration/otp/index.test.ts#L59-L68

Reply:
Well I wanted to test "messageParameters" separately and then just use the "waitForOtpCode" method everywhere, but I see your point